src/table/components/head/HeadCellContent.tsx

```diff
@@ -6,25 +6,38 @@ import { FullSortDirection } from '../../constants';
 import getHeadIcons from '../../utils/get-head-icons';
 import { VisuallyHidden, StyledSortButton, StyledHeadCellContent } from './styles';
 import HeadCellMenu from './HeadCellMenu';
+import { handleHeadKeyDown } from '../../utils/handle-keyboard';
 
 function HeadCellContent({ children, column, isActive, areBasicFeaturesEnabled }: HeadCellContentProps) {
-  const { constraints, selectionsAPI, keyboard, translator, changeSortOrder } = useContextSelector(
+  const { constraints, keyboard, translator, rootElement, changeSortOrder } = useContextSelector(
     TableContext,
     (value) => value.baseProps
   );
+  const setFocusedCellCoord = useContextSelector(TableContext, (value) => value.setFocusedCellCoord);
   const isFocusInHead = useContextSelector(TableContext, (value) => value.focusedCellCoord[0] === 0);
+  const isInSelectionMode = useContextSelector(TableContext, (value) => value.selectionState.colIdx > -1);
```

Comment:
Why could you not use `selectionsAPI.isModal()`?

Reply:
because the header is memoed and does not update when you start selecting. But the selectionState on the context does. We need to change the tabindex of the header once you start selecting. This was the simplest way I could think of

Reply:
What do you mean by "memoed"? The component is not wrapped in `React.memo`? I think you mean `selectionsAPI` "property" is not updated when you start selecting?

You MIGHT be able to do something like:
```js
const isInSelectionMode = useContextSelector(TableContext, (value) => value.baseProps.selectionsAPI.isModal());
```

So that `selectionsAPI.isModel()` is still used everywhere we need to check if we are in "selection mode".

Reply:
well the head wrapper component is memoed, so this one is indirectly. But looking at the selectionsAPI straight on the context is better. Then I think I can test it better as well!